client/components/calypso-i18n-provider/index.tsx

```diff
@@ -4,10 +4,12 @@
 import * as React from 'react';
 import { setLocaleData as setWpI18nLocaleData } from '@wordpress/i18n';
 import { I18nProvider } from '@automattic/react-i18n';
+import { LocaleProvider } from '@automattic/i18n-utils';
 import i18n from 'i18n-calypso';
 
 const CalypsoI18nProvider: React.FunctionComponent = ( { children } ) => {
 	const [ localeData, setLocaleData ] = React.useState( i18n.getLocale() );
+	const [ localeSlug ] = React.useState( i18n.getLocaleSlug() );
```

Comment:
We don't need `useState` here since we're never updating the locale slug.

```suggestion
	const localeSlug = i18n.getLocaleSlug();
```

Although it does look like we're running some code in response to `i18n.on( 'change', ... );`, should be be updating locale slug in response to this?